autosklearn/automl.py

```diff
@@ -783,17 +801,17 @@ def fit(
             time_left_for_ensembles = max(0, self._time_for_task - elapsed_time)
             proc_ensemble = None
             if time_left_for_ensembles <= 0:
-                # Fit only raises error when ensemble_size is not zero but
+                # Fit only raises error when an ensemble class is given but
                 # time_left_for_ensembles is zero.
-                if self._ensemble_size > 0:
+                if self._ensemble_class is not None:
                     raise ValueError(
                         "Not starting ensemble builder because there "
                         "is no time left. Try increasing the value "
                         "of time_left_for_this_task."
                     )
```

Comment:
I didn't really notice this before but does this mean we can end up where a user calls fit but because of bad time management on autosklearn's part, the user gets a `ValueError`? It would make more sense to make this a `RuntimeError` but maybe it's not worth changing at this point.

Reply:
No, they should get an ensemble with a DummyEstimator.